packages/next/pages/_document.js

```diff
@@ -202,7 +203,7 @@ export class Head extends Component {
               name="viewport"
               content="width=device-width,minimum-scale=1,initial-scale=1"
             />
-            <link rel="canonical" href={page.split('.amp')[0]} />
+            <link rel="canonical" href={page.replace('/index','').replace('.amp', '')} />
```

Comment:
`/index` would match the first `/index` in a string, what if the string is like `/index/index/index`?

Reply:
Good point, updated